### ISSUE TITLE
New checker for when an index-lookup is used together with ``dict.items()``

### ISSUE THREAD

**Issue by yushao2** [PyCQA/pylint#4470]
### Is your feature request related to a problem? Please describe
As discussed with @cdce8p in #4445 :

A checker can be implemented to detect for instances where an index-lookup is used with ``dict.items()``

```python
for k10, v in b_dict.items():
  val = v + 2
  print(b_dict[k10])  # this is bad
```

### Describe the solution you'd like

A new checker should be implemented to discourage use of index-lookup when the value can be pulled from ``v``
```python
for k10, v in b_dict.items():
  val = v + 2
  print(b_dict[k10])  # [use-value-from-dict-items]
```
### Additional context

> We could also think about what should happen if an index-lookup is used together with `dict.items`.
> 
> ```python
> for k10, v in b_dict.items():
>     val = v + 2
>     print(b_dict[k10])
> ```
> 
> --
> I realize this might all be a bit much at once, but keep in mind: The work you have already done is really good! Especially for a first-time contributor  These are now just the finishing touches. If you encounter any issues or need help, please don't hesitate to ask.

@cdce8p,

I think this should be in a different error message altogether -- should I try my hand at implementing it within this PR or perhaps I should create a separate PR for that. I think it might be better to not bloat up this PR

_Originally posted by @yushao2 in https://github.com/PyCQA/pylint/issues/4445#issuecomment-837112857_

**Comment by yushao2** [comment]
@cdce8p please assign this to me, i'll work on this alongside str-partition checker
